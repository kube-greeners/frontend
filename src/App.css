@import "~antd/dist/antd.css";

body {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #ececec;
}

#root {
  margin:auto;
}

.flexcolumn {
  display: flex;
  flex-direction: column;
  row-gap: 20px;
}

.ant-card {
  border-radius: 8px;
}

.layout-grid {
  align-content: center;
  height: 100vh;
  display: grid;
  grid-template-columns: 2fr 1fr;
  gap:1rem;
  grid-template-areas: 
    "lc b1"
    "lc b2"
    "lc b3"
    "lc b4";
}
.container{
  position:relative;
  margin-left:auto;
  margin-right:auto;
  padding-right:15px;
  padding-left:15px
}
@media (min-width: 476px) {
  .container {
    padding-right: 15px;
    padding-left: 15px;
  }
}
@media (min-width: 768px) {
  .container {
    padding-right: 15px;
    padding-left: 15px;
  }
}
@media (min-width: 992px) {
  .container {
    padding-right: 15px;
    padding-left: 15px;
  }
}
@media (min-width: 1200px) {
  .container {
    padding-right: 15px;
    padding-left: 15px;
  }
}
@media (min-width: 476px) {
  .container {
    width: 100%;
  }
}
@media (min-width: 768px) {
  .container {
    width: 720px;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .container {
    width: 960px;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .container {
    width: 1140px;
    max-width: 100%;
  }
}
@media (min-width: 1400px) {
  .container {
    width: 1340px;
    max-width: 100%;
  }
}
