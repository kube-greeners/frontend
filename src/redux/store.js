import { configureStore } from '@reduxjs/toolkit'
import dashboardReducer from './dashboardSlice'
import { apiSlice } from './apiSlice'

export default configureStore({
  reducer: {
    dashboard: dashboardReducer,
    [apiSlice.reducerPath]:apiSlice.reducer
  },
  middleware: getDefaultMiddleware =>
  getDefaultMiddleware().concat(apiSlice.middleware)
})