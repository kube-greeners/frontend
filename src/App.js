import { useSelector, useDispatch } from 'react-redux'
import { useEffect, useRef } from 'react';
import { Col, Row, Card } from 'antd'
import Co2Emission from './Components/CO2Emission'
import './App.css';
import { fetchActivePods, fetchCpuUsage, fetchCpuAllocation, fetchMemoryUsage, fetchMemoryAllocation } from './Utilities/dataFetching';


function App() {

  const statusCpuUsage = useSelector(state => state.dashboard.cpu.statusUsage);
  const statusCpuAllocation = useSelector(state => state.dashboard.cpu.statusAllocation);
  const statusMemoryUsage = useSelector(state => state.dashboard.memory.statusUsage);
  const statusMemoryAllocation = useSelector(state => state.dashboard.memory.statusAllocation);
  const statusPods = useSelector(state => state.dashboard.pods.status);

  const fetchingCpuUsageRef = useRef(false)
  const fetchingCpuAllocationRef = useRef(false)
  const fetchingMemoryUsageRef = useRef(false)
  const fetchingMemoryAllocationRef = useRef(false)
  const fetchingPods = useRef(false)

  const active_pods = useSelector(state => state.dashboard.pods.currentValue);
  const cpu_usage = useSelector(state => state.dashboard.cpu.currentUsage);
  const cpu_allocation = useSelector(state => state.dashboard.cpu.currentAllocated);
  const memory_usage = useSelector(state => state.dashboard.memory.currentUsage);
  const memory_allocation = useSelector(state => state.dashboard.memory.currentAllocated);



  const dispatch = useDispatch();

  const statContainerStyle = {
    flex: '1', 
    textAlign: 'center',
    fontWeight: 'bold'
  }

  useEffect(() => {
    const namespace = "production"
    const interval = "10d"
    const step = "1h"

    //Make sure we only fetch the data once. 
    //TODO: check edge cases for failed when we have the correct endpoint
    if (statusCpuUsage === 'idle' && !fetchingCpuUsageRef.current) {
      dispatch(fetchCpuUsage({ namespace, interval, step }))
      fetchingCpuUsageRef.current = true;
    }
    if (statusCpuAllocation === 'idle' && !fetchingCpuAllocationRef.current) {
      dispatch(fetchCpuAllocation({ namespace, interval, step }))
      fetchingCpuAllocationRef.current = true;
    }
    if (statusPods === 'idle' && !fetchingPods.current) {
      dispatch(fetchActivePods({ namespace, interval, step }))
      fetchingPods.current = true;
    }
    if (statusMemoryUsage === 'idle' && !fetchMemoryUsage.current) {
      dispatch(fetchMemoryUsage({ namespace, interval, step }))
      fetchMemoryUsage.current = true;
    }
    if (statusMemoryAllocation === 'idle' && !fetchingMemoryAllocationRef.current) {
      dispatch(fetchMemoryAllocation({ namespace, interval, step }))
      fetchingMemoryAllocationRef.current = true;
    }

  }, [dispatch, statusCpuAllocation, statusCpuUsage, statusPods, statusMemoryUsage, statusMemoryAllocation])

  return (
    <div className="container">
      <div className="layout-grid">
        <Card style={{ height: '100%', gridArea: 'lc' }} title="Estimated CO2 emission"><Co2Emission /> </Card>
        <Card style={{...statContainerStyle,gridArea:'b1'}} title="Saved Emission">Loading...</Card>
        <Card style={{...statContainerStyle,gridArea:'b2'}} title="CPU Usage and Allocation">
          {(statusCpuUsage === 'succeeded' && statusCpuAllocation === `succeeded`) ?
            `${cpu_usage.toFixed(2)} core / ${cpu_allocation.toFixed(2)} core` :
            `Loading...`}
        </Card>
        <Card style={{...statContainerStyle,gridArea:'b3'}} title="Memory Usage  and Allocation">
          {(statusMemoryAllocation === 'succeeded' && statusMemoryUsage === 'succeeded') ?
            `${memory_usage.toFixed(2)} GB / ${memory_allocation.toFixed(2)} GB` :
            'Loading...'}
        </Card>
        <Card style={{...statContainerStyle,gridArea:'b4'}} title="Active Pods">
          {(statusPods === 'succeeded') ?
            active_pods :
            'Loading...'}
        </Card>
      </div>
    </div>
  );
}

export default App;
