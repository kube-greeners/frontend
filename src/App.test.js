import App from "./App";
import renderer from 'react-test-renderer'
import store from "./redux/store";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

test("matches snapshot", () => {
  const tree = renderer.create(<Provider store={store}><App/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
