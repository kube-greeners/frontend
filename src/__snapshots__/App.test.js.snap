// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches App snapshot 1`] = `
Array [
  <nav
    className="navbar"
  >
    <div
      className="logo"
    >
      <svg
        fill="none"
        height="22"
        viewBox="0 0 120 22"
        width="120"
        xmlns="http://www.w3.org/2000/svg"
      >
        <ellipse
          cx="1.95669"
          cy="4.81699"
          fill="#666666"
          rx="1.41176"
          ry="1.42649"
        />
        <ellipse
          cx="1.95669"
          cy="11.474"
          fill="#666666"
          rx="1.41176"
          ry="1.42649"
        />
        <ellipse
          cx="1.95669"
          cy="18.131"
          fill="#666666"
          rx="1.41176"
          ry="1.42649"
        />
        <ellipse
          cx="8.54458"
          cy="4.81699"
          fill="#666666"
          rx="1.41176"
          ry="1.42649"
        />
        <ellipse
          cx="8.54458"
          cy="11.474"
          fill="#666666"
          rx="1.41176"
          ry="1.42649"
        />
        <ellipse
          cx="8.54458"
          cy="18.131"
          fill="#666666"
          rx="1.41176"
          ry="1.42649"
        />
        <ellipse
          cx="15.1334"
          cy="4.81699"
          fill="#666666"
          rx="1.41176"
          ry="1.42649"
        />
        <ellipse
          cx="15.1334"
          cy="11.474"
          fill="#666666"
          rx="1.41176"
          ry="1.42649"
        />
        <ellipse
          cx="15.1334"
          cy="18.131"
          fill="#666666"
          rx="1.41176"
          ry="1.42649"
        />
        <path
          clipRule="evenodd"
          d="M62.5801 0.390503H61.7768V1.23464H62.5801V0.390503ZM55.228 2.09508H54.458V6.40691H55.2525V4.43787C55.2477 4.24934 55.2642 4.06087 55.3016 3.87633C55.4577 3.26192 55.8839 2.74231 56.4907 2.74231C57.1291 2.74231 57.2116 3.26192 57.2116 3.82528V6.40691H58.0061V4.4324C58.0033 4.23202 58.0227 4.03198 58.0639 3.83622C58.2113 3.25463 58.6462 2.74231 59.2285 2.74231C59.8511 2.74231 59.9651 3.21999 59.9651 3.82528V6.40691H60.7614V3.64661C60.7614 2.53994 60.3264 1.99298 59.3916 1.99298C58.7616 2.00389 58.1912 2.38252 57.9166 2.97203H57.9008C57.7605 2.3248 57.3554 1.99298 56.6538 1.99298C56.036 2.01239 55.4828 2.3958 55.228 2.98115H55.2122C55.2225 2.87636 55.2278 2.77108 55.228 2.66574V2.09508ZM61.7856 2.09511H62.5801V6.40694H61.7856V2.09511ZM66.0019 3.7141H65.8265C65.1302 3.7141 63.3272 3.77427 63.329 5.19635C63.329 6.05689 64.0182 6.50904 64.7321 6.50904C65.2972 6.52154 65.8169 6.18828 66.0597 5.65762H66.0755C66.0647 5.77091 66.0594 5.8847 66.0597 5.99855V6.40694H66.7964V3.67946C66.7964 2.5892 66.2106 1.99302 65.1583 1.99302C64.6023 1.98679 64.059 2.16567 63.6078 2.50351L63.9358 3.10881C64.2728 2.85372 64.6751 2.70841 65.0916 2.6913C65.616 2.6913 66.0019 2.93013 66.0019 3.63753V3.7141ZM64.9093 5.84391C65.581 5.84391 66.016 5.11463 66.016 4.47287V4.33613H65.8108C65.2127 4.33613 64.1306 4.37989 64.1306 5.15474C64.1306 5.50479 64.3919 5.84391 64.9093 5.84391ZM59.0005 8.52561H54.4808V21.3809H56.2153V16.6278H59.0005C61.2612 16.6278 62.8905 14.9888 62.8905 12.5494C62.8905 10.11 61.2612 8.52561 59.0005 8.52561ZM58.7216 15.0617C60.2106 15.0617 61.1226 14.0972 61.1226 12.5493C61.1226 11.0015 60.2106 10.0917 58.7391 10.0917H56.2153V15.0617H58.7216ZM66.0071 8.52561H64.3076V18.8631C64.3076 21.1767 65.7282 21.4483 66.6911 21.4483C66.8787 21.4476 67.0661 21.4355 67.2523 21.4119V19.8458C67.1534 19.8575 67.0538 19.8636 66.9542 19.864C66.428 19.864 66.0071 19.6653 66.0071 18.5531V8.52561ZM73.3296 15.627H73.7154V15.4629C73.7154 13.9515 72.8928 13.441 71.7721 13.441C70.4041 13.441 69.3009 14.3344 69.3009 14.3344L68.5994 13.0417C69.5617 12.3187 70.7216 11.9354 71.9089 11.9478C74.1521 11.9478 75.4166 13.224 75.4166 15.5541V21.3883H73.8382V20.515C73.8375 20.2714 73.8492 20.028 73.8733 19.7857H73.8382C73.3213 20.9224 72.2096 21.6366 71.0004 21.6089C69.4763 21.6089 67.9873 20.6444 67.9873 18.8048C67.9873 15.7546 71.8405 15.627 73.3296 15.627ZM73.7329 17.2533C73.7329 18.6188 72.8051 20.1849 71.3687 20.1849V20.1795C70.2655 20.1795 69.7043 19.452 69.7043 18.71C69.7043 17.0527 72.0159 16.9615 73.2962 16.9615H73.7329V17.2533ZM77.9193 13.7143H76.7635V12.294H77.972V9.63582H79.6188V12.294H81.7568V13.7143H79.6188V17.8128C79.6188 19.6506 80.8641 19.8876 81.5112 19.8876C81.6465 19.8893 81.7817 19.8771 81.9146 19.8511V21.4172C81.7168 21.4426 81.5176 21.4548 81.3183 21.4537C80.1801 21.4537 77.9193 21.0891 77.9193 18.0133V13.7143ZM82.8424 13.7143H83.9456L83.9491 21.3808H85.631V13.7143H87.7357V12.294H85.631V11.8929C85.631 10.1828 86.8395 9.964 87.4884 9.964C87.6353 9.96255 87.782 9.97475 87.9268 10.0005V8.48904C87.7125 8.45075 87.4954 8.43245 87.2779 8.43435C86.1537 8.43435 83.9456 8.81539 83.9456 11.8565V12.294H82.8424V13.7143ZM93.1429 11.9478C95.7877 11.9478 97.9257 13.9697 97.9257 16.7555C97.9257 19.5413 95.7877 21.5997 93.1429 21.5997C90.4981 21.5997 88.3619 19.5614 88.3619 16.7555C88.3619 13.9496 90.4981 11.9478 93.1429 11.9478ZM96.191 16.7554C96.191 18.6861 94.8247 20.0699 93.1427 20.0699C91.4783 20.0699 90.0945 18.6697 90.0945 16.7554C90.0945 14.841 91.4608 13.4736 93.1427 13.4736C94.8247 13.4736 96.191 14.8246 96.191 16.7554ZM99.5865 12.1664H101.233V13.769C101.233 14.0004 101.222 14.2317 101.198 14.4618H101.233C101.635 13.1145 102.67 12.057 104.04 12.057C104.186 12.0581 104.332 12.0702 104.476 12.0935V13.8419C104.315 13.8065 104.151 13.7882 103.987 13.7873C102.901 13.7873 101.902 14.5895 101.498 15.955C101.35 16.4936 101.28 17.0522 101.289 17.6123V21.3808H99.5865V12.1664ZM107.456 12.1665H105.809L105.802 21.3809H107.5V17.1748C107.489 16.7714 107.524 16.3681 107.605 15.9734C107.938 14.6625 108.85 13.5503 110.147 13.5503C111.513 13.5503 111.688 14.6625 111.688 15.864V21.3827H113.393V17.1767C113.385 16.7478 113.426 16.3195 113.514 15.9004C113.83 14.6443 114.759 13.5503 116.003 13.5503C117.336 13.5503 117.581 14.5713 117.581 15.864V21.3809H119.284V15.4811C119.284 13.1328 118.355 11.9478 116.359 11.9478C115.01 11.9695 113.788 12.7801 113.202 14.0426H113.167C112.869 12.6588 111.992 11.9478 110.504 11.9478C109.119 11.9478 107.875 12.987 107.456 14.0608H107.419C107.443 13.8375 107.456 13.6129 107.456 13.3881V12.1665Z"
          fill="#666666"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M36.3418 12.9827V9.01568L39.7761 7.03308L43.2122 9.01568V12.9827L39.7761 14.9671L36.3418 12.9827ZM41.9467 12.3528V9.66602L39.7742 8.32882L37.6075 9.66602V12.3384L39.7742 13.6756L41.9467 12.3528Z"
          fill="#E06565"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M40.1387 19.627V15.66L43.5748 13.6755L47.0091 15.66V19.627L43.5748 21.6096L40.1387 19.627ZM45.7435 16.3034L43.5749 14.9672V14.9693L41.4043 16.3034V18.9716L43.5749 20.3181L45.7435 18.984V16.3034Z"
          fill="#1890FF"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M32.5449 19.627V15.66L35.9801 13.6755L39.4153 15.66V19.627L35.9801 21.6096L32.5449 19.627ZM38.1497 16.3034L35.9801 14.9672V14.9693L33.8105 16.3034V18.9716L35.9801 20.3181L38.1497 18.984V16.3034Z"
          fill="#5FA37E"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M43.9355 12.9827V9.01568L47.3708 7.03308L50.806 9.01568V12.9827L47.3708 14.9671L43.9355 12.9827ZM49.5403 9.66602L47.3697 8.32468V8.32882L45.201 9.66602V12.3384L47.3697 13.6756L49.5403 12.3528V9.66602Z"
          fill="#614F8C"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M32.5449 2.3731L35.9811 0.390503L39.4153 2.3731V6.34013L35.9811 8.32456L32.5449 6.34013V2.3731ZM38.1497 3.02504L35.9811 1.68216L33.8105 3.02504V5.69428L35.9811 7.03303L38.1497 5.71081V3.02504Z"
          fill="#E98E56"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="ant-row"
      style={
        Object {
          "marginLeft": -12,
          "marginRight": -12,
          "paddingTop": "7rem",
        }
      }
    >
      <div
        className="ant-col ant-col-7"
        style={
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          }
        }
      >
        <label
          htmlFor="Namespace"
          style={
            Object {
              "color": "#666666",
              "display": "block",
              "marginBottom": ".5rem",
            }
          }
        >
          Namespace
        </label>
        <div
          className="ant-select ant-select-single ant-select-show-arrow ant-select-loading"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <div
            className="ant-select-selector"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="ant-select-selection-search"
            >
              <input
                aria-activedescendant="Namespace_list_0"
                aria-autocomplete="list"
                aria-controls="Namespace_list"
                aria-haspopup="listbox"
                aria-owns="Namespace_list"
                autoComplete="off"
                className="ant-select-selection-search-input"
                id="Namespace"
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onPaste={[Function]}
                readOnly={true}
                role="combobox"
                style={
                  Object {
                    "opacity": 0,
                  }
                }
                type="search"
                unselectable="on"
                value=""
              />
            </span>
            <span
              className="ant-select-selection-item"
              title="All namespaces"
            >
              All namespaces
            </span>
          </div>
          <span
            aria-hidden={true}
            className="ant-select-arrow ant-select-arrow-loading"
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <span
              aria-label="loading"
              className="anticon anticon-loading anticon-spin"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="loading"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <div
        className="ant-col ant-col-7"
        style={
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          }
        }
      >
        <label
          htmlFor="Resource"
          style={
            Object {
              "color": "#666666",
              "display": "block",
              "marginBottom": ".5rem",
            }
          }
        >
          Resource
        </label>
        <div
          className="ant-select ant-select-single ant-select-show-arrow"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <div
            className="ant-select-selector"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="ant-select-selection-search"
            >
              <input
                aria-activedescendant="Resource_list_0"
                aria-autocomplete="list"
                aria-controls="Resource_list"
                aria-haspopup="listbox"
                aria-owns="Resource_list"
                autoComplete="off"
                className="ant-select-selection-search-input"
                id="Resource"
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onPaste={[Function]}
                readOnly={true}
                role="combobox"
                style={
                  Object {
                    "opacity": 0,
                  }
                }
                type="search"
                unselectable="on"
                value=""
              />
            </span>
            <span
              className="ant-select-selection-item"
              title="Estimated CO2 emission"
            >
              Estimated CO2 emission
            </span>
          </div>
          <span
            aria-hidden={true}
            className="ant-select-arrow"
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <span
              aria-label="down"
              className="anticon anticon-down ant-select-suffix"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <div
        className="ant-col ant-col-7"
        style={
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          }
        }
      >
        <label
          style={
            Object {
              "color": "#666666",
              "display": "block",
              "marginBottom": ".5rem",
            }
          }
        >
          Time interval
        </label>
        <div
          className="ant-picker ant-picker-range"
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <div
            className="ant-picker-input ant-picker-input-active"
          >
            <input
              autoComplete="off"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              placeholder="Start date"
              readOnly={true}
              size={12}
              value=""
            />
          </div>
          <div
            className="ant-picker-range-separator"
          >
            <span
              aria-label="to"
              className="ant-picker-separator"
            >
              <span
                aria-label="swap-right"
                className="anticon anticon-swap-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="swap-right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            className="ant-picker-input"
          >
            <input
              autoComplete="off"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              placeholder="End date"
              readOnly={true}
              size={12}
              value=""
            />
          </div>
          <div
            className="ant-picker-active-bar"
            style={
              Object {
                "left": 0,
                "position": "absolute",
                "width": 0,
              }
            }
          />
          <span
            className="ant-picker-suffix"
          >
            <span
              aria-label="calendar"
              className="anticon anticon-calendar"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="calendar"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                />
              </svg>
            </span>
          </span>
          <span
            className="ant-picker-clear"
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <span
              aria-label="close-circle"
              className="anticon anticon-close-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <div
      className="layout-grid"
    >
      <div
        className="ant-card ant-card-bordered"
        style={
          Object {
            "gridArea": "lc",
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              Estimated CO2 emission
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "height": 500,
                "justifyContent": "center",
              }
            }
          >
            <div
              className="lds"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-card ant-card-bordered"
        style={
          Object {
            "flex": "1",
            "fontWeight": "bold",
            "gridArea": "b1",
            "textAlign": "center",
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              Saved Emission
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          Loading...
        </div>
      </div>
      <div
        className="ant-card ant-card-bordered"
        style={
          Object {
            "flex": "1",
            "fontWeight": "bold",
            "gridArea": "b2",
            "textAlign": "center",
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              CPU Usage and Allocation
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          Loading...
        </div>
      </div>
      <div
        className="ant-card ant-card-bordered"
        style={
          Object {
            "flex": "1",
            "fontWeight": "bold",
            "gridArea": "b3",
            "textAlign": "center",
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              Memory Usage  and Allocation
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          Loading...
        </div>
      </div>
      <div
        className="ant-card ant-card-bordered"
        style={
          Object {
            "flex": "1",
            "fontWeight": "bold",
            "gridArea": "b4",
            "textAlign": "center",
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              N Active Pod
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          Loading...
        </div>
      </div>
    </div>
  </div>,
]
`;
